PushNotifications.md

```diff
@@ -80,11 +77,18 @@ To get push notifications setup in your own app, read [Setting up your own app](
 
 Devices need to be [activated](#device-activation) with Ably once. Once activated, you can use their device ID, client ID or push token (APNs device token/ FCM registration token) to push messages to them using the Ably dashboard or a [Push Admin](https://ably.com/documentation/general/push/admin) (SDKs which provide push admin functionality, such as [Ably-java](https://github.com/ably/ably-java), [Ably-js](https://github.com/ably/ably-js), etc.). However, to send push notifications through Ably channels, devices need to [subscribe to a channel for push notifications](#subscribing-to-channels-for-push-notifications). Once subscribed, messages on that channel with a [push payload](#sending-messages) will be sent to devices which are subscribed to that channel.
 
+### Ensure `WidgetsBinding` is initialized
+
+If calling any methods from the `ably_flutter` package in your `main` function, call `WidgetsFlutterBinding.ensureInitialized();`. 
+You only need to do this if calling the functions before calling `runApp()`.
+This is done to ensure all platform methods will be successfully received by the native Ably plugin on the host platform.
+
 ### Device activation
 
 - Create a rest or realtime client: e.g. `final realtime = ably.Realtime(options: clientOptions);`
 - Activate the device for push notifications with Ably: `ablyClient.push.activate();`. This only
   needs to be done once, and will be used across all future app launches, as long as the app is not deactivated. This method will throw an AblyException if it fails.
+- The future returned by `activate` is not guaranteed to complete quickly. For example, in the cases where there is no internet connection, `activate` will wait until internet connection. Therefore, any code waiting for the future to complete should be not important, e.g. logging code.
```

Comment:
```suggestion
- The future returned by `activate` is not guaranteed to complete quickly. For example, if there is no internet connection, `activate` will wait until one is available. Therefore, code waiting for the future to complete should not be important, e.g. logging code.
```

Reply:
It's unclear to me whether you are saying that logging code _is_ important and therefore _not_ a good candidate for this. I assume not?

Reply:
Clarified in fd50412e271b63b00b16edb9c852c0033606678e